app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,14 @@
+class SessionsController < ApplicationController
+  def create
+    if request.env['omniauth.auth']
```

Comment:
so this is kind of a red flag to where logic inside the controller should be simple. In order to accomplish this, you can attempt to apply different type of patterns:

[Service Objects](https://hackernoon.com/service-objects-in-ruby-on-rails-and-you-79ca8a1c946e)
[Form Objects](https://www.codementor.io/victor_hazbun/complex-form-objects-in-rails-qval6b8kt)
[Interactors](https://goiabada.blog/interactors-in-ruby-easy-as-cake-simple-as-pie-33f66de2eb78)